src/counters.c

```diff
@@ -406,13 +406,8 @@ static void *StatsMgmtThread(void *arg)
     SCLogDebug("stats_thread_data %p", &stats_thread_data);
 
     TmThreadsSetFlag(tv_local, THV_INIT_DONE | THV_RUNNING);
+    TmThreadsWaitForUnpause(tv_local);
```

Comment:
should we avoid entering the while loop if this returned early due to the kill flag being set?